### PR TITLE
Add 'form-control-range' for range input type

### PR DESCRIPTION
**Pull request by Oviglo**
| Q             | A
| ------------- | ---
| Branch?       | 5.x
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

Add 'form-control-range' for range input type.

### CODE REVIEW

**Comment by Oviglo** [comment]
I finished my 4 month old pull request: https://github.com/symfony/symfony/pull/38625
Sorry for long time.

**Comment by Oviglo** [comment]
> Hey!
> 
> Excellent, just like I would have done it.
> 
> I think @AngelFQC has recently worked with this code. Maybe they can help review this?
> 
> Cheers!
> 
> Carsonbot

Yes there are some failed builds and I don't know to fix that (it's my first pull request)

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig [hunk 1]
```diff
@@ -135,7 +135,13 @@
...
 {% block form_widget_simple -%}
     {% if type is not defined or type != 'hidden' %}
-        {%- set attr = attr|merge({class: (attr.class|default('') ~ (type|default('') == 'file' ? ' custom-file-input' : ' form-control'))|trim}) -%}
+        {% set className = ' form-control' %}
+        {% if type|default('') == 'file' %}
+            {% set className = ' custom-file-input' %}
+        {% elseif type|default('') == 'range' %}
+            {% set className = ' form-control-range' %}
+        {% endif %}
+        {%- set attr = attr|merge({class: (attr.class|default('') ~ className)|trim}) -%}
```
Comment (AngelFQC) [inline src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig:144]: Change the conditional to one line with ternary operator
```twig
        {%- set attr = attr|merge({class: (attr.class|default('') ~ (type|default('') == 'file' ? ' custom-file-input' : (type|default('') == 'range' ? ' form-control-range' : ' form-control')))|trim}) -%}
```
Or apply the `-` operator to trim spaces and newlines
```twig
        {%- set className = ' form-control' -%}
        {%- if type|default('') == 'file' -%}
            {%- set className = ' custom-file-input' -%}
        {%- elseif type|default('') == 'range' -%}
            {%- set className = ' form-control-range' -%}
        {%- endif -%}
```

**Review by AngelFQC — CHANGES_REQUESTED** [review]
This fails when generating trailing spaces, for example here:
```
Expected
-'<input type="text" id="text" name="text" disabled="disabled" required="required" readonly="readonly" maxlength="10" pattern="\d+" class="foobar form-control" data-foo="bar" value="value" />'
Actual
+'                <input type="text" id="text" name="text" disabled="disabled" required="required" readonly="readonly" maxlength="10" pattern="\d+" class="foobar form-control" data-foo="bar" value="value" />'
```

File: src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig (hunk 1 above)
Comment (Oviglo) [reply src/Symfony/Bridge/Twig/Resources/views/Form/bootstrap_4_layout.html.twig:144]: Thank you

**Comment by AngelFQC** [comment]
@Oviglo maybe you need follow the instructions to rebase your code according to https://symfony.com/doc/current/contributing/code/pull_requests.html#step-4-submit-your-pull-request and pass the fabbot.io check

**Comment by Oviglo** [comment]
[https://github.com/symfony/symfony/pull/40472](https://github.com/symfony/symfony/pull/40472)

**Pull request closed without merging**